lib/models/Author.js

```diff
@@ -0,0 +1,51 @@
+const pool = require('../utils/pool');
+
+class Author {
+  id;
+  name;
+  books;
+  constructor(row) {
+    this.id = row.id;
+    this.name = row.name;
+    this.books = row.books ?? [];
+  }
+
+  static async insert({ name }) {
+    const { rows } = await pool.query(
+      'INSERT INTO authors (name) VALUES ($1) RETURNING *',
+      [name]
+    );
+    return new Author(rows[0]);
+  }
+
+  async addBookById(bookId) {
+    await pool.query(
+      'INSERT INTO author_books (author_id, book_id) VALUES ($1, $2) RETURNING *',
+      [this.id, bookId]
+    );
+    return this;
+  }
+
+  static async getAll() {
+    const { rows } = await pool.query('SELECT * from authors');
+    return rows.map((row) => new Author(row));
+  }
+
+  static async getById(id) {
+    const { rows } = await pool.query(
+      `SELECT 
+      authors.*, 
+      COALESCE(
+        json_agg(to_jsonb(books))
+        FILTER (WHERE books.id IS NOT NULL), '[]'
+    ) as books from authors 
+      LEFT JOIN author_books on author.id = author_books.author_id 
```

Comment:
I think here on line 42 the `author.id` needs to be `authors.id` since the table is plural. This should fix your getById issue